Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,13 +524,53 @@ bool isFunctionJson(string fullName)
             }
         }
 
-        private async Task<string> DeployZipLocally(ZipDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        private async Task ArtifactFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var artifactDeploymentInfo = (ArtifactDeploymentInfo)deploymentInfo;
+            //Fetches content from URL if ArtifactURL exists
+            HttpContent content = !string.IsNullOrEmpty(artifactDeploymentInfo.ArtifactURL)
+                ? await DeploymentHelper.GetArtifactContentFromURL(artifactDeploymentInfo, tracer)
+                : Request.Content;
+
+            //repository.RepositoryPath is the directory the deployment pipeline expects the files that will be deployed to be
+            var tempDirPath = repository.RepositoryPath;
+            if (deploymentInfo.CleanupTargetDirectory)
+            {
+                var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+                if (targetInfo.Exists)
+                {
+                    // If tempDirPath already exists, rename it so we can delete it concurrently with
+                    // the unzip (along with any other junk in the folder)
+                    var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                    using (tracer.Step(string.Format("Renaming extractTargetDirectory({0}) to tempDirectory({1})", targetInfo.FullName, moveTarget)))
+                    {
+                        targetInfo.MoveTo(moveTarget);
+                    }
+                }
+
+            }
+
+            var tempFilePath = Path.Combine(tempDirPath, deploymentInfo.FileName);
+            Directory.CreateDirectory(tempDirPath);
+            using (tracer.Step("Saving request content to {0}", tempFilePath))
+            {
+                await content.CopyToAsync(tempFilePath, tracer);
```

Comment:
As per you comment that you want to do copying and deleting concurrently, this should be:
Task copyTask = content.CopyToAsync();
Task cleanTask = Task.Run(() => DeleteFilesAndDirsExcept());
Task.WaitAll(cleanTask, extractTask);